src/lib/helpers/wish/roll.js

```diff
@@ -82,7 +82,7 @@ const roll = async (banner, indexOfBanner, WishInstance) => {
 	}
 	result.fateType = result.rarity === 3 ? 'stardust' : 'starglitter';
 	result.fateQty = result.isNew
-		? 0
+		? rarity === 4 ? 2 : 10
```

Comment:
Is it really like this?

![image](https://user-images.githubusercontent.com/13815468/224527606-57c13e7e-5734-4d88-9ef9-8735b0dcd890.png) 
![image](https://user-images.githubusercontent.com/13815468/224528221-da69cbf9-57d2-40f1-bc95-7d5debd07ace.png)

I have read the description one more time, I think this doesn't seem right either. What I understand are:
- Starglitter will be obtained from **duplicate characters** depends on the char constellation
- Obtaining weapon will provide starglitter/stardust even on the first copy

Reply:
With that change, getting the first copy of a character still gives you no starglitter, for some reason
Oh but there could be a problem, let me check what getting a character over c7 does

Reply:
![image](https://user-images.githubusercontent.com/118117643/224538323-51817ba5-4e57-482a-a609-6866916a1f32.png)
![image](https://user-images.githubusercontent.com/118117643/224538332-6bcc4d8e-ebdc-4a45-96ba-c5b542f49659.png)
First copies give no starglitter. This also applies to 5 stars.
------------------------------------------
![image](https://user-images.githubusercontent.com/118117643/224538227-c4755faf-ddc2-46c9-8e05-948e5d517aae.png)
C7+ characters give 5/25 Starglitter depending on rarity
------------------------------------------
![image](https://user-images.githubusercontent.com/118117643/224538466-d55ceb5c-bec4-4460-8c2c-56c844ca55eb.png)
![image](https://user-images.githubusercontent.com/118117643/224538477-6fc18c60-86e7-4aad-bdd9-6b059fc7a92d.png)
Weapons always give Starglitter

Reply:
![image](https://user-images.githubusercontent.com/118117643/224546440-7221c2e0-c019-4a7b-945f-33d5df571b15.png)

You were right, there was a problem. 3 star weapons were giving me 10 (or whatever was after the " : " in line 85) stardust instead of the standard 15, so I added a bit of code to say "if you're 3 star, give me 15; if you're 4 star, give me 2; if you're 5 star, give me 10". I suppose it's line 83 that's telling the app to give stardust instead of starglitter?

There's probably a more elegant way to put it into code, but this is all I've got unfortunately haha
I'll make another separate pull request once I'm feeling more confident that it works, but so far, characters and weapons are giving me the right starglitter AND stardust values.

Reply:
This code still doesn't seem right, the value of stardust & starglitter already processed by `getMilestoneQty` function,
and this code will still give you starglitter on the first acquired character, it shouldn't be like that.

The only task we need to do is giving stardust & starglitter to every weapon obtained even in the first copy. Actually I have a solution for this. you can verify it first below.
![image](https://user-images.githubusercontent.com/13815468/224558705-8e357046-de48-4ff8-8eda-3b343c832c75.png)

Reply:
Got it. Then I guess my code was working only because that function was taking priority, giving the correct 0 Starglitter on first copies. Anyways, thank you for your time, and I'm glad you managed to find a solution a lot cleaner and readable than mine :)

One last question. Once you receive a pull request as the owner of a project, are you always forced to merge everything or nothing, or you can select which changes to take in and what to leave out? 

Because I don't know if, when proposing multiple changes, I should create multiple separate forks with one change each, or a single fork with many changes, and then you just decide what to merge.

Reply:
I can't merge a specific files from pull request, it will merge the whole branch, so I recommend to create multiple pull request for different issues.
An issue can contain multiple changes and commits

If there any files that need to be changed, I can ask you to modify them first before I merge them.